src/server/chromium/crNetworkManager.ts

```diff
@@ -382,6 +404,8 @@ class InterceptableRequest implements network.RouteDelegate {
   private _client: CRSession;
   _timestamp: number;
   _wallTime: number;
+  private _extraInfoCallbacks: (() => void)[] = [];
+  private _extraInfoReceived: boolean = false;
```

Comment:
suggestion: substitute callbacks / boolean flag to a promise with continuation and the following methods:

```js
private _extraInfoPromise;
private _extraInfoPromiseFulfill;

async ensureExtraInfo() { await this._extraInfoPromise; }

applyExtraInfo(event) {
  this.request.updateWithRawHeaders(headersObjectToArray(event.headers));
  this._extraInfoPromiseFulfill.call(null);
}

maybeDisregardExtraInfo(esponsePayload: Protocol.Network.Response) {
  this._extraInfoPromiseFulfill.call(null);
}
```